### DISCUSSION TITLE
Is it possible to cache widgets (similarly to IndexedStack/AutomaticKeepAliveClientMixin)?

### DISCUSSION THREAD

**Discussion by manuel-plavsic** [rrousselGit/riverpod#1393] [Q&A] [answered]
I would like to cache a widget.

My app is structured as follows:

- 4 `BottomNavigationBarItem`s
  - Inside the scaffold of the first `BottomNavigationBarItem`: `TabBar` and `TabBarView` (partially already cached with `AutomaticKeepAliveClientMixin`).
  
The `TabBar` caches the content as long as I don't leave the widget (scaffold) containing the `TabBar` and `TabBarView`. Unfortunately, as soon as I tap on one of the other `BottomNavigationBarItems` (which, in my case, changes `GoRouter` route) and come back to the one I just left, the information regarding tab-wise scrolling position and last selected tab are lost.

Can Riverpod help me prevent losing this cache?

**Comment by rrousselGit** [comment, marked as answer]
You can use AutomaticKeepAliveClientMixin/IndexedStack if you want

Riverpod is unrelated to this
